lib/src/controllers/player_controller.dart

```diff
@@ -114,26 +163,30 @@ class PlayerController extends ChangeNotifier {
   ///
   /// See also:
   /// * [FinishMode]
-  Future<void> startPlayer({FinishMode finishMode = FinishMode.stop}) async {
+  Future<void> startPlayer({
+    FinishMode finishMode = FinishMode.stop,
+    bool forceRefresh = true,
+  }) async {
     if (_playerState == PlayerState.initialized ||
         _playerState == PlayerState.paused) {
       final isStarted = await AudioWaveformsInterface.instance
           .startPlayer(_playerKey.toString(), finishMode);
       if (isStarted) {
-        setPlayerState(PlayerState.playing);
+        _setPlayerState(PlayerState.playing);
       } else {
         throw "Failed to start player";
       }
     }
+    setRefresh(forceRefresh);
```

Comment:
Here, setRefresh method calls notifyListeners. So, no need to notify it again. Or you can add conditions on when to notify.

Reply:
Or you can create 2 different Methods, `setRefresh` and `_setRefresh`. and implementation of setRefresh will be 

```dart
void setRefresh() {
_setRefresh();
notifyListeners();
}
```

and _setRefresh will update the data. You can use _setRefresh inside of this player controller and other users can use setRefresh for manually updating value. Here, _setRefresh will not notify listeners.